### PR TITLE
add enterprise downloads page

### PR DESCRIPTION
**Pull request by jescalan**
Adds a page specifically for downloading enterprise versions. All enterprise versions require a license key in order to run.

### CODE REVIEW

File: website/pages/downloads/enterprise.jsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+              </code>
+            </div>
+            <div className={s.releaseCandidate}>
+              <p>
+                A beta for Consul v1.10.0 is available! The release can be{' '}
+                <a href="https://releases.hashicorp.com/consul/1.10.0-beta3/">
+                  downloaded here
+                </a>
+              </p>
+            </div>
```
Comment (blake) [inline website/pages/downloads/enterprise.jsx:87]: This block was removed on the regular download page by commit https://github.com/hashicorp/consul/commit/fbd7acc7545a67ae19659472035a5761f5730eba  in the main branch.

**Review by blake — APPROVED** [review]
LGTM. I had one minor comment about removing the beta callout.

**Comment by blake** [comment]
Hey @jescalan, do you still want to merge this PR?

**Comment by jescalan** [comment]
@blake - no not yet, still working with a few folks to figure out whether it's needed or not. sorry for the delays!

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/537233.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 32a2cf4646c7602eccf58f3067fe0d0d826a8abc onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 32a2cf4646c7602eccf58f3067fe0d0d826a8abc onto `release/1.11.x` succeeded!

**Pull request merged**